src/components/EnhancedTable/EnhancedTable.tsx

```diff
@@ -0,0 +1,99 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { AutoSizer } from 'react-virtualized';
+import MuiTable, { IMuiVirtualizedTableProps, IMuiVirtualizedTableColumn } from 'mui-virtualized-table';
+import naturalSort = require('javascript-natural-sort');
+
+import Loading from '../Loading';
+
+export interface IProps<TRow> extends Omit<IMuiVirtualizedTableProps<TRow>, 'width'> {
+  loading?: boolean;
+  defaultOrderBy?: string;
+  defaultOrderDirection?: 'asc' | 'desc';
+  width?: number;
+}
+
+export function EnhancedTable<TRow>({
+  loading,
+  defaultOrderBy,
+  defaultOrderDirection,
+  data,
+  width: propWidth,
+  height: propHeight,
+  maxHeight,
+  ...tableProps
+}: IProps<TRow>) {
+  type SortParams = { orderBy: keyof TRow; orderDirection?: 'asc' | 'desc' };
+  const [sortParams, setSortParams] = React.useState<SortParams | null>(
+    defaultOrderBy
+      ? {
+          orderBy: defaultOrderBy as keyof TRow,
+          orderDirection: defaultOrderDirection
+        }
+      : null
+  );
+
+  const sortedData = React.useMemo(() => {
+    if (!sortParams) {
+      return data;
+    }
+
+    const { orderBy, orderDirection } = sortParams;
+
+    return [...data].sort((first, second) => {
+      const a = first[orderBy];
+      const b = second[orderBy];
+      if (_.isNil(a)) {
+        return 1;
+      }
+      if (_.isNil(b)) {
+        return -1;
+      }
+
+      if (orderDirection === 'desc') {
+        return naturalSort(_.toString(b), _.toString(a));
+      } else {
+        return naturalSort(_.toString(a), _.toString(b));
+      }
+    });
+  }, [data, sortParams]);
+
+  const onHeaderClick = React.useCallback((col: IMuiVirtualizedTableColumn<TRow>) => {
```

Comment:
this no longer takes into account disableSort

Reply:
Hmm good point. lemme check

Reply:
Ah right, so `disableSort` was placed on the column but that causes all kinds of type issues. I'll have to figure out another way to restore that functionality.